tests/integration/tools/client/src/main/java/io/helidon/tests/integration/tools/client/TestClient.java

```diff
@@ -0,0 +1,327 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.tests.integration.tools.client;
+
+import java.io.StringReader;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonException;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import javax.json.JsonValue;
+
+import io.helidon.media.common.MessageBodyReadableContent;
+import io.helidon.media.jsonp.JsonpSupport;
+import io.helidon.webclient.WebClient;
+import io.helidon.webclient.WebClientRequestBuilder;
+
+import org.junit.jupiter.api.Assertions;
+
+/**
+ * Web client to access all services of remote test application.
+ */
+public class TestClient {
+
+    private static final Logger LOGGER = Logger.getLogger(TestClient.class.getName());
+
+    private static final String UTF_8_STR = StandardCharsets.UTF_8.toString();
+
+    private static String encode(final String str) {
+        try {
+            return URLEncoder.encode(str, UTF_8_STR);
+        } catch (UnsupportedEncodingException ex) {
+            Assertions.fail(String.format("URL fragment encoding failed: %s", ex.getMessage()));
+        }
+        return "";
+    }
+
+    private static String logStackTrace(final JsonObject response) {
+        JsonArray stacktrace = response.getJsonArray("stacktrace");
+        List<JsonObject> tracesList = stacktrace.getValuesAs(JsonObject.class);
+        String logMsg = null;
+        for (JsonObject trace : tracesList) {
+            String message = trace.getString("message");
+            LOGGER.warning(() -> String.format("%s: %s", trace.getString("class"), trace.getString("message")));
+            JsonArray lines = trace.getJsonArray("trace");
+            List<JsonObject> linesList = lines.getValuesAs(JsonObject.class);
+            linesList.forEach((line) -> {
+                LOGGER.warning(() -> String.format("    at %s$%s (%s:%d)",
+                        line.getString("class"), line.getString("method"), line.getString("file"), line.getInt("line")));
+            });
+            if (logMsg == null) {
+                logMsg = message;
+            }
+        }
+        return logMsg;
+    }
+
+    /**
+     * Remote test web client builder.
+     */
+    public static class Builder {
```

Comment:
Inner classes should be last within the class. Please change order.

Ordering should be:
constants
static fields
final fields
fields
constructor(s)
static method(s)
public methods
package local methods
protected methods
private methods
public inner classes
package local inner classes
private inner classes

Reply:
Order fixed.